src/main/java/roomescape/service/MemberService.java

```diff
@@ -0,0 +1,53 @@
+package roomescape.service;
+
+import roomescape.controller.dto.LoginRequest;
+import roomescape.controller.dto.MemberRequest;
+import roomescape.controller.dto.MemberResponse;
+import roomescape.domain.Member;
+import roomescape.exception.ErrorCode;
+import roomescape.exception.RoomEscapeException;
+import roomescape.repository.MemberRepository;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class MemberService {
+
+	private final MemberRepository memberRepository;
+
+	MemberService(MemberRepository memberRepository) {
+		this.memberRepository = memberRepository;
+	}
+
+	public MemberResponse create(MemberRequest request) {
+		var member = Member.builder().name(request.name()).email(request.email()).password(request.password()).build();
+
+		var isExists = this.memberRepository.isExists(request.name());
+		if (isExists) {
+			throw new RoomEscapeException(ErrorCode.DUPLICATE_MEMBER);
+		}
+		else {
+			var savedMember = this.memberRepository.save(member);
+			return MemberResponse.from(savedMember);
+		}
+	}
+
+	public MemberResponse findMemberByLoginRequest(LoginRequest request) {
+		var findedMember = this.memberRepository.findByEmail(request.email());
+
+		if (findedMember != null) {
+			checkPassword(request.password(), findedMember.getPassword());
+			return MemberResponse.from(findedMember);
+		}
+		else {
+			throw new RoomEscapeException(ErrorCode.NOT_FOUND_MEMBER);
+		}
```

Comment:
```suggestion
        var findedMember = this.memberRepository.findByEmail(request.email())
            .orElseThrow(() -> new RoomEscapeException(ErrorCode.NOT_FOUND_MEMBER));

        checkPassword(request.password(), findedMember.getPassword());
        return MemberResponse.from(findedMember);
```

repository 에서 optional 을 반환한다면 null 체크도 수월해지고 조금 더 관리하기 쉬운 로직이 만들어질 것 같아요.
유근님은 어떻게 생각하시나요?

Reply:
`optional` 를 사용하면 null 체크가 간결해지고 가독성이 높아지는 장점과 NullPointerException 의 발생 가능성을 줄여줄 것 같아요
근데, `optional` 객체를 생성하는데 추가 메모리가 필요해서 성능 오버헤드가 발생할 수 있을 것 같아요
`optional` 을 남용하거나 잘못 사용하면 코드의 복잡성이 증가 할 수 있을 것 같아요
그래서 개인적으로 `optional` 을 지양하기는 해요

이런 문서도 있으니 참고하시면 좋을 것 같아요~
https://dzone.com/articles/using-optional-correctly-is-not-optional

이 부분에 대해서는 조금 더 고민해볼게요 !

Reply:
일단 성능 오버헤드는 그렇게 크게 고려하진 않아도 될 것 같아요.
정말 척박한(?) 환경에서 서버가 돌아가지 않는 이상 그 정도는 견딜 수 있을 것 같거든요!
그리고 남용하고 잘못 사용하면 복잡성이 증가하는 것에 저도 굉장히 동의해요.
하지만 위의 예시정도는...! 어느정도 괜찮은 범위에 있다고 생각합니다 ㅋㅋㅋ

글을 읽어보니 어느정도 공감이 되는 내용도 있고, 조금 극단적이라고 생각이 드는 item 도 있네요.
하지만 아예 사용하지 않는 것을 정하고 개발 및 유지보수를 하시는거라면 저도 딱히 반대할 이유는 없는 것 같아요!

Reply:
저도 해당 코드에서는 사용해도 나쁘지는 않을 것 같다는 생각은 해요 ㅋㅋㅋ
다른 더 좋은 방법이 없을까 고민 좀 더 해보면서 다음 단계에 적용해볼게요! 
물론, 척박한 환경에서 서버가 돌아가진 않을텐데 문제가 생기면 의심하는 포인트를 줄이고 싶은 목적이 크긴해요 ㅋㅋ